### ISSUE TITLE
ApplicationListeners between ConfigFileApplicationListener and LoggingApplicationListener

### ISSUE THREAD

**Issue by cemo** [spring-projects/spring-boot#2543]
I need to set a property before Logging System initialised.  

```
ConfigFileApplicationListener: Ordered.HIGHEST_PRECEDENCE + 10;
LoggingApplicationListener   : Ordered.HIGHEST_PRECEDENCE + 11;
```

My use case: I am using Logback with placeholder `${variable}` support. I would like to set a property to be set at my logback configuration file. 

Please change orders to be something we can put other listeners between them.

**Comment by philwebb** [comment]
Seems sensible but I'm not keen to change this on a point release in case someone is depending on the current value. The `LoggingApplicationListener` does have a `setOrder` method, so perhaps it's possible to use that in the meantime?

**Comment by cemo** [comment]
Thanks, I missed it. Order is instance variable in `LoggingApplicationListener`. How can I access and configure it?  


By the way I have noticed that Logback is configured twice. One for default configuration and after Boot is resetting configuration and configuring again. I do not know how this can be achieved but It would be better to configure just once. I know that there are some necessary things have to be done before logging configuration. But this still looks like debatable on my side. I would like to open a new issue for this as well, if this OK for your.

**Comment by philwebb** [comment]
The listeners get added to `SpringApplication` in the `initialize()` method. You should be able to get to them after the application is created but before run is called:

```
SpringApplication application = new SpringApplication(MyApp.class);
application.getListeners(); // The LoggingListener should be in here
application.run(args);
```

**Comment by philwebb** [comment]
Please raise another issue for the Logback configured twice problem.
